### PR TITLE
[flatex] Implement foreign currencies parsing for buy, sell and dividend documents. Ignore saving plan confirmation.

### LINKED ISSUE #599: FlatEx Dividende(Fondsertragausschuettung) von Realty Income
Ich hab noch 26 weitere Fondsertragsabrechnungen die werde ich jetzt nicht auch noch hochladen werde weil... ich bin faul, 
das scheint aber ein problem mit REITs (in meinm Fall Realty Income, LTC Properties und STORE Capital) zu sein da es sich bei den 26 stk alles um Dividendenzahlungen von eben jenen sind .

Mfg

## Document
<details><summary>20200316_Fondsertragsausschuettung_1012727978_145613110.pdf</summary><p>

```json
[
  [
    "6602266811107161",
    "5946472414769339",
    "flatex Bank AG",
    "Postfach 100551",
    "41405 Neuss",
    "USt-IdNr.: DE 246 786 363",
    "Kundenservice:",
    "Tel.: +49 (0)9221 - 7035898",
    "E-Mail: kunden@flatex.de",
    "flatex Bank AG -  Rotfeder-Ring 7 - 60327 Frankfurt am Main",
    "3814548783081106901630",
    "16.03.2020",
    "Kzode",
    "Pdohyt Jnövrols",
    "BG Hmvtmbhbzl",
    "Glavjgfwe 8",
    "97053 Iüuqglygoeq pw Pape",
    "Ertragsmitteilung - ausschüttender/teilthesaurierender Fonds",
    "Lokp Xvuvkxoygbg: 9576527923",
    "Swmhygzlhkda    : Oeöekrbp, Kzyyvq",
    "Nr.1838121276                  REALTY INC. CORP.    DL 1 (US7561091049/899744)",
    "St.             :             3,00    Bruttoausschüttung",
    "pro Stück          :       0,2325000 USD",
    "Extag           :       28.02.2020    Bruttoausschüttung :            0,70 USD",
    "Valuta          :       13.03.2020",
    "Bemessungsgrundlage:            0,00 EUR",
    "Devisenkurs     :         1,110400   *Einbeh. Steuer     :            0,00 EUR",
    "Quellenst.-satz :            15,00 %  Gez. Quellensteuer :            0,10 USD",
    "Endbetrag          :            0,54 EUR",
    "* Enthalten sind folgende Steuern (negative Werte bedeuten Steuererstattung)",
    "Kapitalertragsteuer  :            0,00 EUR",
    "Solidaritätszuschlag :            0,00 EUR",
    "Kirchensteuer        :            0,00 EUR",
    "Details dazu inkl. evtl. angerechneter ausländischer Quellensteuer finden",
    "Sie im Steuerreport unter der Transaktion-Nr.: 1838121276  .",
    "Die Verrechnung des Endbetrag erfolgt über Ihr Konto Nr.: 1012727978.",
    "Fondsinformationen:",
    "steuerpflichtiger Ertrag                                 :            0,70 USD",
    "davon ausländische Erträge mit anrechenb. QSt         :            0,00 USD",
    "ausländische Erträge mit anrechenb. fikt. QSt   :            0,00 USD",
    "anrechenbare ausländische Quellensteuer**                :            0,10 USD",
    "**  ggf. mit einbeh. Steuer verrechnet und/oder Quellensteuer-Topf zugeführt",
    "Die  Gutschrift erfolgt  unter Vorbehalt  des  Eingangs.  Einwendungen  müssen",
    "unverzüglich  nach  Zugang  erhoben  werden.  Die  Unterlassung  rechtzeitiger",
    "Einwendung gilt als Genehmigung.",
    "Dieser   Beleg   ist    keine   Steuerbescheinigung.    Kapitalerträge   sind",
    "einkommensteuerpflichtig.",
    "Diese Mitteilung ist maschinell erstellt und wird nicht unterschrieben.",
    "Für weitergehende Fragen wenden Sie sich bitte an Ihr flatex-Service-Team.",
    "BLZ 101 308 00 / BIC: BIWBDE33XXX  -  Aktiengesellschaft, Sitz Frankfurt  -  Amtsgericht Frankfurt am Main (HRB 105687)",
    "Vorsitzender des Aufsichtsrats: Martin Korbmacher  -  Vorstand: Frank Niehage (Vorsitzender), Jörn Engelmann"
  ]
]

```

</p></details>

## Community profile

https://community.tresor.one/u/70c5b913

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #640: Flatex: Fehler bei der Importierung von Dividenden
## Document
<details><summary>Royal Dutch Shell A.pdf</summary><p>

```json
[
  [
    "9697617669157002",
    "2858167587147839",
    "lxvprl Ifqq LM",
    "Flvxbwvt 782103",
    "55217 Ntdhf",
    "UHp-DaDk.: IP 661 034 966",
    "Iwtocvtfigdge:",
    "Tjf.: +56 (3)9838 - 3392317",
    "M-Otby: xmakfr@mgafvu.rl",
    "flatex Bank AG -  Rotfeder-Ring 7 - 60327 Frankfurt am Main",
    "8608174899966443495606",
    "Saxpysdtz, 80.96.3178",
    "Oezvu",
    "Oraplhh Sj",
    "Hkobwandl. 14",
    "63747 Sjgimvjxb",
    "Dividendengutschrift für ausländische Wertpapiere",
    "Yjjh Ejwubzsqunb: 6084352726",
    "Qujkkguidzfq    : Ht, Hqioucp",
    "Nr.2028188022  ROYAL DUTCH SHELL A EO-07               (GB00B03MLX29/A0D94M)",
    "St.             :           25,00      Bruttodividende",
    "pro Stück       :          0,1353 EUR",
    "Extag           :      13.08.2020      Bruttodividende :            3,38 EUR",
    "Zahlungstag     :      22.09.2020      Bemessungs-",
    "Valuta          :      22.09.2020      grundlage       :            0,00 EUR",
    "*Einbeh. Steuer  :            0,00 EUR",
    "Quellenst.-satz :           15,00 %    Gez. Quellenst. :            0,51 EUR",
    "Endbetrag       :            2,87 EUR",
    "* Enthalten sind folgende Steuern (negative Werte bedeuten Steuererstattung)",
    "Kapitalertragsteuer  :            0,00 EUR",
    "Solidaritätszuschlag :            0,00 EUR",
    "Kirchensteuer        :            0,00 EUR",
    "Details dazu inkl. evtl. angerechneter ausländischer Quellensteuer",
    "finden Sie im Steuerreport unter der Transaktion-Nr. : 2028188022",
    "Die Verrechnung des Endbetrages erfolgt über Ihr Konto Nr.: 1015976960",
    "Anrechenbare Quellensteuer**                  15,00 %  =            0,51 EUR",
    "** nur relevant für Steuerinländer",
    "Die  Gutschrift erfolgt  unter Vorbehalt  des  Eingangs.  Einwendungen  müssen",
    "unverzüglich  nach  Zugang  erhoben  werden.  Die  Unterlassung  rechtzeitiger",
    "Einwendung gilt als Genehmigung.",
    "Dieser   Beleg   ist    keine   Steuerbescheinigung.    Kapitalerträge   sind",
    "einkommensteuerpflichtig.",
    "Diese Mitteilung ist maschinell erstellt und wird nicht unterschrieben.",
    "Für weitergehende Fragen wenden Sie sich bitte an Ihr flatex-Service-Team.",
    "BLZ 101 308 00 / BIC: BIWBDE33XXX  -  Aktiengesellschaft, Sitz Frankfurt  -  Amtsgericht Frankfurt am Main (HRB 105687)",
    "Vorsitzender des Aufsichtsrats: Martin Korbmacher  -  Vorstand: Frank Niehage (Vorsitzender), Jörn Engelmann, Steffen Jentsch"
  ]
]

```

</p></details>

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #651: [FlatEx] Spar-/Entnahmepläne
## Document
<details><summary>20201107_BestaetigungNeuanlageAenderungZahlungsplaene_1014175944_171023011.pdf</summary><p>

```json
[
  [
    "0113304030000101",
    "2000001710230110",
    "flatex Bank AG",
    "Postfach 100551",
    "41405 Neuss",
    "USt-IdNr.: DE 246 786 363",
    "Kundenservice:",
    "Tel.: +49 (0)9221 - 7035898",
    "E-Mail: kunden@flatex.de",
    "flatex Bank AG -  Rotfeder-Ring 7 - 60327 Frankfurt am Main",
    "0304032000001710230110",
    "Frankfurt, 07.11.2020",
    "Herrn",
    "Max Mustermann",
    "Muster Str. 3",
    "01234 FlatExCity",
    "Ihr Auftrag - Spar-/Entnahmepläne - vom 06.11.2020",
    "Ihre Depotnummer: 1014175951",
    "Depotinhaber:     Max Mustermann",
    "------------------------------------------------------------------------------",
    "Einrichtung Sparplan Nr. 0001111111",
    "------------------------------------------------------------------------------",
    "ISIN           : IE00B3VVMM84         Bezeichnung: VANGUARD FTSE EMU.ETF DLD",
    "Rate           : 50,00 EUR",
    "Start          : 01.12.2020",
    "Ende           : 30.06.2021",
    "Termin         : 30.06.2021 / 30.06.2021",
    "Die Verrechnung der Endbeträge erfolgt über Ihr Konto Nr. 1111111111.",
    "------------------------------------------------------------------------------",
    "______________________________________________________________________________",
    "Seite 1/1",
    "BLZ 101 308 00 / BIC: BIWBDE33XXX  -  Aktiengesellschaft, Sitz Frankfurt  -  Amtsgericht Frankfurt am Main (HRB 105687)",
    "Vorsitzender des Aufsichtsrats: Martin Korbmacher  -  Vorstand: Frank Niehage (Vorsitzender), Jörn Engelmann, Steffen Jentsch"
  ]
]

```

</p></details>

## Community profile

https://app.tresor.one/p/5fe5983d7393bb0031399f26

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #736: [Flatex] Dividendenausschüttung in USD nicht erkannt
Der Endbetrag ist in EUR.
Aktuell wird das PDF nicht mehr erkannt. In der Vergangenheit wurde es erkannt, aber leider falsch. Es wurde immer die Bemessungsgrundlage als Summe genommen.
Dieses Mal ist die Bemessungsgrundlage 0 EUR (warum auch immer).

Ich erwarte die Bruttodividende als Summe in Tresor.one. Diese ist hier als USD angegeben, also muss man sie noch umrechnen mit dem angegeben Devisenkurs. Bruttoausschüttung/Devisenkurs.

## Document
<details><summary>20201201_Fondsertragsausschuettung_11_22.pdf</summary><p>

```json
[
  [
    "6766957383409906",
    "1671011832479119",
    "jfkyvw Mqpp MR",
    "Zrggvzwc 954252",
    "79268 Tnwve",
    "AGn-OcPb.: EL 319 150 524",
    "Kundenservice:",
    "Tel.: +49 (0)9221 - 7035898",
    "E-Mail: kunden@flatex.de",
    "flatex Bank AG -  Rotfeder-Ring 7 - 60327 Frankfurt am Main",
    "0102262000001734436500",
    "Frankfurt, 01.12.2020",
    "Jqurm",
    "Bwohjru Rhnaia",
    "Owdsrxmvv. 73/8",
    "64085 Xmkfdhijhe",
    "Ertragsmitteilung - ausschüttender/teilthesaurierender Fonds",
    "Doit Hrpqflqdrbb: 3245886960",
    "Vhgcuykdfvbq    : Ihypmp, Yoqoery",
    "Nr.2102869656                LTC PTIES INC.     DL-,01 (US5021751020/884625)",
    "St.             :            15,00    Bruttoausschüttung",
    "pro Stück          :       0,1900000 USD",
    "Extag           :       19.11.2020    Bruttoausschüttung :            2,85 USD",
    "Valuta          :       30.11.2020",
    "Bemessungsgrundlage:            0,00 EUR",
    "Devisenkurs     :         1,198000   *Einbeh. Steuer     :            0,00 EUR",
    "Quellenst.-satz :            15,00 %  Gez. Quellensteuer :            0,43 USD",
    "Endbetrag          :            2,02 EUR",
    "* Enthalten sind folgende Steuern (negative Werte bedeuten Steuererstattung)",
    "Kapitalertragsteuer  :            0,00 EUR",
    "Solidaritätszuschlag :            0,00 EUR",
    "Kirchensteuer        :            0,00 EUR",
    "Details dazu inkl. evtl. angerechneter ausländischer Quellensteuer finden",
    "Opj nl Qulqgiahuaht qwrkn wip Siicbymnilr-Tx.: 1417675404.",
    "Qht Mkiqzoqlohj bgo Nyksiddos zkhfgue üznk Yzp Ldpuk Ew.: 1291393164.",
    "Fondsinformationen:",
    "steuerpflichtiger Ertrag                                 :            2,85 USD",
    "davon ausländische Erträge mit anrechenb. QSt         :            0,00 USD",
    "ausländische Erträge mit anrechenb. fikt. QSt   :            0,00 USD",
    "anrechenbare ausländische Quellensteuer**                :            0,43 USD",
    "**  ggf. mit einbeh. Steuer verrechnet und/oder Quellensteuer-Topf zugeführt",
    "Die  Gutschrift erfolgt  unter Vorbehalt  des  Eingangs.  Einwendungen  müssen",
    "unverzüglich  nach  Zugang  erhoben  werden.  Die  Unterlassung  rechtzeitiger",
    "Einwendung gilt als Genehmigung.",
    "Dieser   Beleg   ist    keine   Steuerbescheinigung.    Kapitalerträge   sind",
    "einkommensteuerpflichtig.",
    "Diese Mitteilung ist maschinell erstellt und wird nicht unterschrieben.",
    "Für weitergehende Fragen wenden Sie sich bitte an Ihr flatex-Service-Team.",
    "BLZ 101 308 00 / BIC: BIWBDE33XXX  -  Aktiengesellschaft, Sitz Frankfurt  -  Amtsgericht Frankfurt am Main (HRB 105687)",
    "Vorsitzender des Aufsichtsrats: Martin Korbmacher  -  Vorstand: Frank Niehage (Vorsitzender), Jörn Engelmann, Steffen Jentsch"
  ]
]

```

</p></details>

## Community profile

https://community.tresor.one/u/41569a41

---
Issue submitted with https://import.tresor.one

### PR DESCRIPTION
**Pull request by dredav**
- Adjust the dividend parsing logic to get all cases with dividend in foreign currencies get parsed. 
- Adjust the buy (and sell) logic for stock prices in foreign currencies (reported by mail).
- Fix #640
- Fix #651
- Fix #599
- Fix #736
- Add sample from #640 as test case
- Add sample from #651 as test case
- Add sample from mail as test case
- Rename mock files
- Add activity support for `fxRate` and `foreignCurrency`
- Ignore create saving plan confirmation with status code `7`

### CODE REVIEW

File: src/brokers/flatex.js [hunk 1]
```diff
@@ -236,64 +237,83 @@ const findDateDividend = (textArr, startLineNumber) => {
...
-    textArr,
+const findPayout = (textArr, startLineNumber, fxRate, foreignCurrency) => {
+  const payoutAmount = Big(
+    parseGermanNum(grossValueByGroupIndex(textArr, startLineNumber, 1))
+  );
+  const payoutCurrency = grossValueByGroupIndex(textArr, startLineNumber, 2);
+
+  if (foreignCurrency === undefined || foreignCurrency !== payoutCurrency) {
+    return payoutAmount;
+  }
```
Comment (SirGibihm) [inline src/brokers/flatex.js:248]: Shouldnt this be: ` if (foreignCurrency === undefined || foreignCurrency === payoutCurrency) {`

File: src/brokers/flatex.js [hunk 2]
```diff
@@ -236,64 +237,83 @@ const findDateDividend = (textArr, startLineNumber) => {
...
+
+  return payoutAmount.div(fxRate);
+};
+
+// GroupIndex = 1: amount, GroupIndex = 2: currency
+const grossValueByGroupIndex = (content, startLineNumber, groupIndex) => {
```
Comment (SirGibihm) [inline src/brokers/flatex.js:254]: This should be witihn the function calling this snippet IMO

File: src/brokers/flatex.js [hunk 3]
```diff
@@ -236,64 +237,83 @@ const findDateDividend = (textArr, startLineNumber) => {
...
-          2
-        );
-      }
+  return undefined;
+};
 
-      if (conversionRate === null) {
-        return parseGermanNum(payoutForeign.split(' ')[0]);
-      }
+const findForeignInformation = (content, startLineNumber) => {
```
Comment (SirGibihm) [inline src/brokers/flatex.js:288]: Maybe a small comment on what this function returns would be good

**Review by SirGibihm — CHANGES_REQUESTED** [review]
I think I found one tiny mistake, thats it. Nice work!

File: src/brokers/flatex.js (hunk 1 above)
Comment (dredav) [reply src/brokers/flatex.js:248]: No. The conversion should only executed when the fxRate currency is the same as the payout currency. But I also need look again in the code to understand this, so I did a minor change and add some comments.

File: src/brokers/flatex.js (hunk 2 above)
Comment (dredav) [reply src/brokers/flatex.js:254]: Sure!

**Comment by SirGibihm** [comment]
Sadly there is a merge confluct. Could you fix that (tried to fix it using the ui but it seems to root deeper than I thought)

**Comment by dredav** [comment]
@SirGibihm Conflicts solved.

**Review by SirGibihm — APPROVED** [review]

**Pull request merged**